baker/GrapherBaker.tsx

```diff
@@ -82,123 +75,32 @@ export const renderDataPageOrGrapherPage = async (
-    if (
-        // This could be folded into the showDataPageOnChartIds check below, but
-        // is kept separate to reiterate that that abscence of a datapageJson leads to
-        // rendering a grapher page
-        !datapageJson ||
-        parseErrors.length > 0 ||
-        // We only want to render datapages on selected charts, even if the
-        // variable found on the chart has a datapage configuration.
-        !grapher.id ||
-        !datapageJson.showDataPageOnChartIds.includes(grapher.id) ||
-        // Fall back to rendering a regular grapher page if the datapage is not
-        // published
-        datapageJson.status !== "published"
-    )
-        return renderGrapherPage(grapher)
-
-    if (!datapageJson.googleDocEditLink)
-        return renderToHtmlPage(
-            <DataPage
-                grapher={grapher}
-                variableId={id}
-                datapageJson={datapageJson}
-                baseUrl={BAKED_BASE_URL}
-                baseGrapherUrl={BAKED_GRAPHER_URL}
-                isPreviewing={false}
-            />
-        )
-
-    // Enrich the datapage JSON with all the published charts this variable is
-    // being used in (aka "related charts") if none have been defined manually.
-    // We also want to exclude from the list the chart that is displayed at the
-    // top of the page to avoid the unnecessary redundancy.
-    datapageJson.allCharts =
-        datapageJson.allCharts ??
-        (await getRelatedChartsForVariable(id, [grapher.id]))
-
-    // Compliment the text-only content from the JSON with rich text from the
-    // companion gdoc, if any
-    const datapageGdoc = await getDatapageGdoc(
-        datapageJson.googleDocEditLink,
-        false,
-        publishedExplorersBySlug
-    )
-
-    // A datapage gdoc is optional. However, if a gdoc is referenced in the
-    // JSON, we assume the gdoc is expected to be used. Additionally, we need
-    // the datapage gdoc to be published so that images in the gdoc content are
-    // baked in bakeGdriveImages() (see also apiRouter.put("/gdocs/:id")). We
-    // also check that the datapage gdoc is of type "fragment" so that it
-    // doesn't get a URL when baked. If any of these conditions is not met, we
-    // log an error and fall back to a grapher page.
-    const dataPageNotPublishedError = `The data page for variable ${datapageJson.title} (${id}) has not been published.`
-    if (!datapageGdoc) {
-        logErrorAndMaybeSendToBugsnag(
-            new JsonError(
-                `${dataPageNotPublishedError} Please check that the googleDocEditLink ${datapageJson.googleDocEditLink} is correct.`
-            )
-        )
-        return renderGrapherPage(grapher)
-    }
+    // If we have a single Y variable and that one has a schema version >= 2,
+    // meaning it has the metadata to render a datapage, then we show the datapage
+    // based on this information. Otherwise we see if there is a legacy datapage.json
```

Comment:
Support for legacy datapage.json configs is dropped, no?

Would also be good to update the code comment above 👆